front/app/containers/ProjectsShowPage/shared/survey/index.tsx

```diff
@@ -218,6 +218,7 @@ class Survey extends PureComponent<Props, State> {
                   requiresConfirmation,
                   flow: 'signup',
                   pathname: window.location.pathname,
+                  modalNoCloseSteps: ['confirmation'],
```

Comment:
Just wondering if it's a module whether adding this string to the array shouldn't happen in the module instead?

Reply:
It's a good comment and it does go agains my previous argument of leakage of module logic into the core. This problem initially surged in the previous user confirmation epic, and would require quite a lot of refactoring to effectively solve. 

I would consider this more of a technical debt separate task if that sounds good to you too

Reply:
I think it wouldn't take a lot of effort. It's going to be comparable to inserting tabs. So you'd keep the steps array in the component's state and insert from the module with useEffect. Unless I'm overlooking something and it's a lot more work, feel free to log. Perhaps you could still log it as a separate task, because a mechanism to insert into an array or object is something we will be need more often in the future and something I was planning to look into soon anyways.